fdbserver/include/fdbserver/DataDistributionTeam.h

```diff
@@ -22,6 +22,35 @@
 
 #include "fdbclient/StorageServerInterface.h"
 
+struct GetTeamRequest;
+namespace data_distribution {
+class EligibilityCounter {
+public:
+	// The type value are used to do bit operations to get combined type. Ex. combineType = LOW_CPU | LOW_DISK_UTIL .
+	// When adding more types, the value should be 2^n, like 4, 8, 16...
+	enum Type {
+		NONE = 0, // don't care about eligibility
+		LOW_CPU = 1,
+		LOW_DISK_UTIL = 2
+	};
+
+	// set the count of type to 0
+	void reset(Type type);
+
+	// return the minimal count of a combined eligible type
+	unsigned getCount(int combinedType) const;
+
+	// increase the count of type
+	void increase(Type type);
+
+	static int fromGetTeamRequest(GetTeamRequest const&);
+
+private:
+	std::map<Type, unsigned int> type_count;
```

Comment:
This is just holding the eligibility for a single team, right? Wouldn't the "count" be either 0 or 1 based on whether or not the team is eligible? When would the count be > 1?

If the count can only be 0 or 1, then this can instead be an `unordered_set` of the `Type`s that are eligible.

Reply:
> Wouldn't the "count" be either 0 or 1 based on whether or not the team is eligible? When would the count be > 1?

In every update round the count will increment if the team is eligible for a specific type, I leave it as a count but not a bool flag on purpose because the count tells how many continuous rounds the team is in low utilization. This makes it possible to extend the eligibility to "give me teams that continue to be low utilized for N rounds" in the future if we meet some oscillation scenarios.